### PR TITLE
Add more annotations to CompletableFuture

### PR DESCRIPTION
**Pull request by ctron**

### CODE REVIEW

File: libraries/java/java/util/concurrent/CompletableFuture.eea [hunk 1]
```diff
@@ -1,7 +1,148 @@
...
+runAsync
+ (Ljava/lang/Runnable;)Ljava/util/concurrent/CompletableFuture<Ljava/lang/Void;>;
+ (L1java/lang/Runnable;)L1java/util/concurrent/CompletableFuture<Ljava/lang/Void;>;
+runAsync
+ (Ljava/lang/Runnable;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture<Ljava/lang/Void;>;
+ (L1java/lang/Runnable;L1java/util/concurrent/Executor;)L1java/util/concurrent/CompletableFuture<Ljava/lang/Void;>;
 supplyAsync
  <U:Ljava/lang/Object;>(Ljava/util/function/Supplier<TU;>;)Ljava/util/concurrent/CompletableFuture<TU;>;
- <U:Ljava/lang/Object;>(Ljava/util/function/Supplier<TU;>;)L1java/util/concurrent/CompletableFuture<TU;>;
+ <U:Ljava/lang/Object;>(L1java/util/function/Supplier<TU;>;)L1java/util/concurrent/CompletableFuture<TU;>;
```
Comment (vorburger) [inline libraries/java/java/util/concurrent/CompletableFuture.eea]: @ctron would you like to amend the commit message to explain why this changes? (Personally I've no idea what it means...)

**Comment by vorburger** [comment]
@maggu2810 you may be interested in reviewing this?

File: libraries/java/java/util/concurrent/CompletableFuture.eea (hunk 1 above)
Comment (maggu2810) [reply libraries/java/java/util/concurrent/CompletableFuture.eea]: * the supplier argument must be non-null
* a non-null completable future is returned

Comment (ctron) [reply libraries/java/java/util/concurrent/CompletableFuture.eea]: You can't really see it from the Javadoc [1], you will always get a non-null result back and you need to provide a non-null supplier.

[1]https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#supplyAsync-java.util.function.Supplier-

Comment (vorburger) [reply libraries/java/java/util/concurrent/CompletableFuture.eea]: IMHO it would be cool if this explanation were in the commit message instead of a PR comment,  just to be more "durable".. because in a future look back one would have to look really hard for to find again, coming from a look at the git history.

Comment (ctron) [reply libraries/java/java/util/concurrent/CompletableFuture.eea]: Sorry it took a bit longer. Just found some time to do that.
